Conversion fails if there are misleading links in /etc/leapp/repos.d

Hello.
I encountered an issue while converting one of the instances.
When I initiated `leapp preupgrade`, it failed with the following error:
```
2024-09-04 19:06:28,395 - INFO - ============================================================
2024-09-04 19:06:28,395 - INFO - 2024-09-04 19:06:26.563154 [ERROR] Actor: dnf_transaction_check
2024-09-04 19:06:28,395 - INFO - Message: Failed to execute script to apply transaction workaround yum config fix. Message: Command ['systemd-nspawn', '-register=no', 'quiet', '-D', u'/var/lib/leapp/scratch/mounts/root_/system_overlay', 'setenv=LEAPP_OVL_SIZE=4096', 'setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.9', 'setenv=LEAPP_NO_RHSM=0', 'setenv=LEAPP_EXPERIMENTAL=0', 'setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', 'setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade;66d7e0d7/common/tools:/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/tools', 'setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade;66d7e0d7/common/files:/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/files', 'setenv=LEAPP_IPU_IN_PROGRESS=7to8', 'setenv=LEAPP_UNSUPPORTED=0', 'setenv=LEAPP_EXECUTION_ID=b5a24ee8-ba5b-46b2-bf0e-0dd6e78cdbe9', '-setenv=LEAPP_HOSTNAME=newvps.skyevents.biz', '/bin/bash', '-c', u'/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/tools/handleyumconfig'] failed with exit code 127.
2024-09-04 19:06:28,395 - INFO - ============================================================
2024-09-04 19:06:28,395 - INFO - END OF ERRORS
2024-09-04 19:06:28,395 - INFO - ============================================================
2024-09-04 19:06:28,395 - INFO - Debug output written to /var/log/leapp/leapp-preupgrade.log
2024-09-04 19:06:28,395 - INFO - ============================================================
2024-09-04 19:06:28,395 - INFO - REPORT OVERVIEW
2024-09-04 19:06:28,396 - INFO - ============================================================
2024-09-04 19:06:28,396 - INFO - Following errors occurred and the upgrade cannot continue:
2024-09-04 19:06:28,396 - INFO - 1. Actor: dnf_transaction_check
2024-09-04 19:06:28,396 - INFO - Message: Failed to execute script to apply transaction workaround yum config fix. Message: Command ['systemd-nspawn', '-register=no', 'quiet', '-D', u'/var/lib/leapp/scratch/mounts/root_/system_overlay', 'setenv=LEAPP_OVL_SIZE=4096', 'setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.9', 'setenv=LEAPP_NO_RHSM=0', 'setenv=LEAPP_EXPERIMENTAL=0', 'setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', 'setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade;66d7e0d7/common/tools:/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/tools', 'setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade;66d7e0d7/common/files:/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/files', 'setenv=LEAPP_IPU_IN_PROGRESS=7to8', 'setenv=LEAPP_UNSUPPORTED=0', 'setenv=LEAPP_EXECUTION_ID=b5a24ee8-ba5b-46b2-bf0e-0dd6e78cdbe9', '-setenv=LEAPP_HOSTNAME=newvps.skyevents.biz', '/bin/bash', '-c', u'/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/tools/handleyumconfig'] failed with exit code 127.
2024-09-04 19:06:28,396 - INFO - HIGH and MEDIUM severity reports:
```
After a brief investigation, I noticed that `leapp` uses `/etc/leapp/repos.d/system_upgrade;66d7e0d7/` instead of `/etc/leapp/repos.d/system_upgrade`. I removed the link and reinstalled `leapp` packages, and now both `leapp preupgrade` and `leapp upgrade` work fine.

I am not sure why `/etc/leapp/repos.d/system_upgrade;66d7e0d7` was created initially. Unfortunately, all traces seem to be lost. However, it might be worthwhile to ensure the dnf workaround works for such paths.
Also, if you have any ideas to prevent the creation of such links, I would be grateful to hear them.

--- a/leapp-data.spec
+++ b/leapp-data.spec
@@ -1,5 +1,7 @@
 %global pes_events_build_date 20240827
 
+%define repositorydir %{_datadir}/leapp-repository/repositories
+
 %define dist_list almalinux centos eurolinux oraclelinux rocky
 %define conflict_dists() %(for i in almalinux centos eurolinux oraclelinux rocky; do if [ "${i}" != "%{dist_name}" ]; then echo -n "leapp-data-${i} "; fi; done)
 
@@ -46,7 +48,7 @@
 
 Name:		leapp-data-%{dist_name}
 Version:	0.4
-Release:	11%{?dist}.%{pes_events_build_date}
+Release:	12%{?dist}.%{pes_events_build_date}
 Summary:	data for migrating tool
 Group:		Applications/Databases
 License:	ASL 2.0
@@ -113,9 +115,9 @@ mv -f %{buildroot}%{_sysconfdir}/leapp/files/leapp_upgrade_repositories.repo.el8
 mv -f %{buildroot}%{_sysconfdir}/leapp/files/repomap.json.el8 \
       %{buildroot}%{_sysconfdir}/leapp/files/repomap.json
 rm -f %{buildroot}%{_sysconfdir}/leapp/files/*.el9
-mkdir -p %{buildroot}%{_sysconfdir}/leapp/repos.d/system_upgrade/common/files/rpm-gpg/8/
+mkdir -p %{buildroot}%{repositorydir}/system_upgrade/common/files/rpm-gpg/8/
 for key in %{gpg_key}; do
-    mv -f files/rpm-gpg/${key} %{buildroot}%{_sysconfdir}/leapp/repos.d/system_upgrade/common/files/rpm-gpg/8/
+    mv -f files/rpm-gpg/${key} %{buildroot}%{repositorydir}/system_upgrade/common/files/rpm-gpg/8/
 done
 %endif
 %if 0%{?rhel} == 8
@@ -124,9 +126,9 @@ mv -f %{buildroot}%{_sysconfdir}/leapp/files/leapp_upgrade_repositories.repo.el9
 mv -f %{buildroot}%{_sysconfdir}/leapp/files/repomap.json.el9 \
       %{buildroot}%{_sysconfdir}/leapp/files/repomap.json
 rm -f %{buildroot}%{_sysconfdir}/leapp/files/*.el8
-mkdir -p %{buildroot}%{_sysconfdir}/leapp/repos.d/system_upgrade/common/files/rpm-gpg/9/
+mkdir -p %{buildroot}%{repositorydir}/system_upgrade/common/files/rpm-gpg/9/
 for key in %{gpg_key}; do
-    mv -f files/rpm-gpg/${key} %{buildroot}%{_sysconfdir}/leapp/repos.d/system_upgrade/common/files/rpm-gpg/9/
+    mv -f files/rpm-gpg/${key} %{buildroot}%{repositorydir}/system_upgrade/common/files/rpm-gpg/9/
 done
 %endif
 
@@ -143,16 +145,19 @@ python3 tests/check_debranding.py %{buildroot}%{_sysconfdir}/leapp/files/pes-eve
 %files
 %doc LICENSE NOTICE README.md
 %if 0%{?rhel} == 8
-%{_sysconfdir}/leapp/repos.d/system_upgrade/common/files/rpm-gpg/9/
+%{repositorydir}/system_upgrade/common/files/rpm-gpg/9/*
 %endif
 
 %if 0%{?rhel} == 7
-%{_sysconfdir}/leapp/repos.d/system_upgrade/common/files/rpm-gpg/8/
+%{repositorydir}/system_upgrade/common/files/rpm-gpg/8/*
 %endif
 %{_sysconfdir}/leapp/files/*
 
 
 %changelog
+* Fri Nov 15 2024 Yuriy Kohut <ykohut@almalinux.org> - 0.4-12.20240827
+- Move operating systems' GPG keys from %{_sysconfdir}/leapp/repos.d to %{repositorydir}
+
 * Mon Oct 14 2024 Yuriy Kohut <ykohut@almalinux.org> - 0.4-11.20240827
 - Support elevation on machines other than x86_64 with adding relevant architectures into map files
 - Back kernelcare vendor support for upgrades from 8 to 9
@@ -215,7 +220,7 @@ python3 tests/check_debranding.py %{buildroot}%{_sysconfdir}/leapp/files/pes-eve
 
 * Tue Jul 16 2024 Andrew Lukoshko <alukoshko@almalinux.org> - 0.2-14.20230823
 - Add CentOS 7 ELS repos support for upgrades to AlmaLinux
- 
+
 * Mon Jul 1 2024 Yuriy Kohut <ykohut@almalinux.org> - 0.2-13.20230823
 - Define 'supported_vendors' and 'target_version' to simplify data management for specific version
 - Support of MariaDB verndors data for both EL8 and EL9

